src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/StackExchangeRedis2Plus/SessionCache.cs

```diff
@@ -19,7 +19,7 @@ public class SessionCache : IStackExchangeRedisCache
     {
         private readonly EventWaitHandle _stopHandle = new EventWaitHandle(false, EventResetMode.ManualReset);
 
-        private readonly ConcurrentDictionary<string, ProfilingSession> _sessionCache = new ConcurrentDictionary<string, ProfilingSession>();
+        private readonly ConcurrentDictionary<ISegment, ProfilingSession> _sessionCache = new ConcurrentDictionary<ISegment, ProfilingSession>();
```

Comment:
Do we ever copy/clone a Segment, where using a reference as the key might be a problem?

Reply:
I don't believe we clone, but we do "combine segments", and that uses the existing segment data with some timing changes.  That looks to be called only after the segment has ended and is being processed in the TransactionTransformer.